routes/transaction.go

```diff
@@ -2547,6 +2547,287 @@ func (fes *APIServer) TransferDAOCoin(ww http.ResponseWriter, req *http.Request)
+	requestData := CreateOrCancelDAOCoinLimitOrderRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(
+			ww,
+			fmt.Sprintf("CreateOrCancelDAOCoinLimitOrder: Problem parsing request body: %v", err),
+		)
+		return
+	}
+
+	if requestData.TransactorPublicKeyBase58Check == "" {
+		_AddBadRequestError(
+			ww,
+			fmt.Sprint("CreateOrCancelDAOCoinLimitOrder: Must provide a TransactorPublicKeyBase58Check."),
+		)
+		return
+	}
+
+	if requestData.SellingDAOCoinCreatorPublicKeyBase58CheckOrUsername == "" &&
+		requestData.BuyingDAOCoinCreatorPublicKeyBase58CheckOrUsername == "" {
+		_AddBadRequestError(
+			ww,
+			fmt.Sprint("CreateOrCancelDAOCoinLimitOrder: Must provide either a "+
+				"BuyingDAOCoinCreatorPublicKeyBase58CheckOrUsername or SellingDAOCoinCreatorPublicKeyBase58CheckOrUsername "+
+				"or both"),
+		)
+		return
+	}
+
+	if !validateExchangeRateCoinsToSellPerCoinToBuyParam(requestData) {
+		_AddBadRequestError(
+			ww,
+			fmt.Sprint("CreateOrCancelDAOCoinLimitOrder: Must provide exactly one of the "+
+				"ScaledExchangeRateCoinsToSellPerCoinToBuy or ExchangeRateCoinsToSellPerCoinToBuy params"),
+		)
+		return
+	}
+
+	if !validateQuantityToBuyParam(requestData) {
+		_AddBadRequestError(
+			ww,
+			fmt.Sprint("CreateOrCancelDAOCoinLimitOrder: Must provide exactly one of the "+
+				"ScaledExchangeRateCoinsToSellPerCoinToBuy or ExchangeRateCoinsToSellPerCoinToBuy params"),
+		)
+		return
+	}
+
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddBadRequestError(
+			ww,
+			fmt.Sprintf("CreateOrCancelDAOCoinLimitOrder: Problem fetching utxoView: %v", err),
+		)
+		return
+	}
+
+	// Decode the transactor public key
+	transactorPublicKeyBytes, _, err := fes.GetPubKeyAndProfileEntryForUsernameOrPublicKeyBase58Check(
+		requestData.TransactorPublicKeyBase58Check,
+		utxoView,
+	)
+	if err != nil {
+		_AddBadRequestError(
+			ww,
+			fmt.Sprintf("CreateOrCancelDAOCoinLimitOrder: Problem decoding transactor public key %s: %v",
+				requestData.TransactorPublicKeyBase58Check, err))
+		return
+	}
+
+	// Compute the additional transaction fees as specified by the request body and the node-level fees.
+	additionalOutputs, err := fes.getTransactionFee(
+		lib.TxnTypeDAOCoinTransfer,
+		transactorPublicKeyBytes,
+		requestData.TransactionFees,
+	)
+	if err != nil {
+		_AddBadRequestError(
+			ww,
+			fmt.Sprintf("CreateOrCancelDAOCoinLimitOrder: TransactionFees specified in Request body are invalid: %v", err),
+		)
+		return
+	}
+
+	buyingCoinPublicKey := lib.ZeroPublicKey.ToBytes()
+	sellingCoinPublicKey := lib.ZeroPublicKey.ToBytes()
+
+	if requestData.BuyingDAOCoinCreatorPublicKeyBase58CheckOrUsername != "" {
+		buyingCoinPublicKey, _, err = fes.GetPubKeyAndProfileEntryForUsernameOrPublicKeyBase58Check(
+			requestData.BuyingDAOCoinCreatorPublicKeyBase58CheckOrUsername,
+			utxoView,
+		)
+		if err != nil {
+			return
```

Comment:
should add some error code here before returning - otherwise user will get a 200 response with no constructed tx